x-pack/plugins/osquery/server/plugin.ts

```diff
@@ -21,6 +22,158 @@ import { initSavedObjects } from './saved_objects';
+        }),
+        privilegeGroups: [
+          {
+            groupType: 'mutually_exclusive',
+            privileges: [
+              {
+                id: 'live_queries_all',
+                includeIn: 'all',
+                name: 'All',
+                savedObject: {
+                  all: [],
+                  read: [],
+                },
+                ui: ['crudLiveQueries', 'readLiveQueries'],
+              },
+              {
+                id: 'live_queries_read',
+                includeIn: 'read',
+                name: 'Read',
+                savedObject: {
+                  all: [],
+                  read: [],
+                },
+                ui: ['readLiveQueries'],
+              },
+            ],
+          },
+          {
+            groupType: 'independent',
+            privileges: [
+              {
+                id: 'saved_queries_only',
+                name: i18n.translate(
+                  'xpack.features.ossFeatures.discoverCreateShortUrlPrivilegeName',
+                  {
+                    defaultMessage: 'Run Saved queries only',
+                  }
+                ),
+                includeIn: 'none',
+                savedObject: {
+                  all: [],
+                  read: [savedQuerySavedObjectType],
+                },
+                ui: ['savedQueriesOnly'],
+              },
+            ],
+          },
+        ],
+      },
+      {
+        name: i18n.translate('xpack.features.ossFeatures.discoverShortUrlSubFeatureName', {
+          defaultMessage: 'Saved queries',
+        }),
+        privilegeGroups: [
+          {
+            groupType: 'mutually_exclusive',
+            privileges: [
+              {
+                id: 'saved_queries_all',
+                includeIn: 'all',
+                name: 'All',
+                savedObject: {
+                  all: [savedQuerySavedObjectType],
+                  read: [savedQuerySavedObjectType],
+                },
+                ui: ['crudSavedQueries', 'readSavedQueries'],
+              },
+              {
+                id: 'saved_queries_read',
+                includeIn: 'read',
+                name: 'Read',
+                savedObject: {
+                  all: [],
+                  read: [savedQuerySavedObjectType],
+                },
+                ui: ['readSavedQueries'],
+              },
+            ],
+          },
+        ],
+      },
+      {
+        name: i18n.translate('xpack.features.ossFeatures.discoverShortUrlSubFeatureName', {
+          defaultMessage: 'Scheduled query groups',
+        }),
+        privilegeGroups: [
+          {
+            groupType: 'mutually_exclusive',
+            privileges: [
+              {
+                id: 'packs_all',
+                includeIn: 'all',
+                name: 'All',
+                savedObject: {
+                  all: [packSavedObjectType],
+                  read: [packSavedObjectType],
```

Comment:
You don't need to grant both `all` and `read` SO privileges for a given type:
```suggestion
                  all: [packSavedObjectType],
                  read: [],
```